File: leetcode/go/672.bulb-switcher-ii.go
func flipLights(n int, presses int) int {
    if n > 3 {
        n = 3
    }
    switch presses {
        case 0:
            return 1
        case 1:
            return []int{2,3,4}[n-1]
        case 2:
            return []int{2,4,7}[n-1]
        default:
            return []int{2,4,8}[n-1]
    }
}

File: leetcode/rust-leetcode/src/672.bulb-switcher-ii.rs
impl Solution {
    pub fn flip_lights(n: i32, presses: i32) -> i32 {
        let n = n.min(3) as usize;
        match presses {
            0 => 1,
            1 => [2,3,4][n-1],
            2 => [2,4,7][n-1],
            _ => [2,4,8][n-1]
        }
    }
}

